map.py

```diff
@@ -0,0 +1,70 @@
+from typing import List
+from .country import Country
+from .city import City
+
+
+class Map:
+    _grid_offset = 10
+
+    def __init__(self, countries_data):
+        self.countries = []
+        self.grid: List[List[City]] = [[None] * (self._grid_offset + 2) for _ in range((10 + 2))]
+        self.__initialize_grid(countries_data)
+        self.__validate_foreign_neighbours()
+
+    def simulate_euro_diffusion(self) -> None:
+        if len(self.countries) == 1:
+            self.countries[0].day_of_full = 0
+            return
+
+        full = False
+        day = 1
+        while not full:
+            
```

Comment:
add 
for x in range(self._grid_offset + 1):
                for y in range(self._grid_offset + 1):
                    if self.grid[x][y] is not None:
                        city = self.grid[x][y]
                        city.transfer_to_neighbours()